Runtime/Core/ApiGatewayManagement/AmazonGameLiftClientWrapper.cs

```diff
@@ -42,17 +62,194 @@ public async Task<SearchGameSessionsResponse> SearchGameSessions(SearchGameSessi
         }
 
-        public async Task<DescribeGameSessionsResponse> DescribeGameSessions(DescribeGameSessionsRequest request)
+        private RetriveAwsCredentialsResponse SetupCredentials()
         {
-            return await _amazonGameLiftClient.DescribeGameSessionsAsync(request);
+            ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
+            var credentials = RetrieveAwsCredentials(_gameLiftClientSettings.ProfileName);
+            return credentials;
+        }
+
+        private async Task GenerateAuthToken()
+        {
+            try
+            {
+                var computeAuthTokenRequest = new GetComputeAuthTokenRequest
+                {
+                    ComputeName = _gameLiftClientSettings.ComputeName,
+                    FleetId = _gameLiftClientSettings.FleetID
+                };
+                var computeAuthTokenResponse = await GetComputeAuthToken(computeAuthTokenRequest);
+                _gameLiftClientSettings.AuthToken = computeAuthTokenResponse.AuthToken;
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                throw;
+            }
+        }
+
+        private async Task RegisterCompute()
+        {
+            //Reregistering compute will just return the same compute back to the user.
+            try
+            {
+                
+                var ipAddress = Dns.GetHostEntry(Dns.GetHostName())
+                    .AddressList
+                    .FirstOrDefault(ip => ip.AddressFamily == AddressFamily.InterNetwork)?
+                    .ToString() ?? "0.0.0.0";
+                
+                var registerComputeRequest = new RegisterComputeRequest()
+                {
+                    ComputeName = _gameLiftClientSettings.ComputeName,
+                    FleetId = _gameLiftClientSettings.FleetID,
+                    IpAddress = ipAddress,
+                    Location = _gameLiftClientSettings.FleetLocation
+                };
+                var registerComputeResponse = await RegisterCompute(registerComputeRequest);
+                    
+                _gameLiftClientSettings.WebSocketUrl = registerComputeResponse.Compute.GameLiftServiceSdkEndpoint;
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+            }
         }
+
+        private async Task CreateCustomLocation()
+        {
+            try
+            {
+                ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
+                var listLocationsResponse = await ListLocations(new ListLocationsRequest
+                {
+                    Filters = new List<string>{ "CUSTOM" }
+                });
+                
+                var foundLocation = listLocationsResponse.Locations.FirstOrDefault(l => l.LocationName.ToString() == _gameLiftClientSettings.FleetLocation);
+                
+                if(foundLocation == null)
```

Comment:
Should we resolve this promise with this `foundLocation` instead? Or are we not planning to implement it since this is a temporary code?

minor: whitespace after `if`

Reply:
As this is temporary until I move it to the UI, I will leave it like this. I will return the `foundLocation` in that system if it makes sense. Otherwise the found location just means that the location that we originally gave it is correct and don't actually need the value, but I could see use for the `LocationModel`.